api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ESMonitoringManager.java

```diff
@@ -67,11 +71,34 @@
         final LocalDateTime oldestMonitoring = oldestMonitoringDate();
         final LocalDateTime start = requestedStart.isAfter(oldestMonitoring) ? requestedStart : oldestMonitoring;
         final LocalDateTime end = Optional.ofNullable(to).orElseGet(DateUtils::nowUTC);
+        final Duration interval = interval(start, end);
         return end.isAfter(start) && end.isAfter(oldestMonitoring)
-                ? getStats(nodeName, start, end)
+                ? getStats(nodeName, start, end, interval)
                 : Collections.emptyList();
     }
 
+    @Override
+    public InputStream getStatsForNodeAsInputStream(final String nodeName,
+                                                    final LocalDateTime from,
+                                                    final LocalDateTime to,
+                                                    final Duration interval) {
+        final LocalDateTime requestedStart = Optional.ofNullable(from).orElseGet(() -> creationDate(nodeName));
+        final LocalDateTime oldestMonitoring = oldestMonitoringDate();
+        final LocalDateTime start = requestedStart.isAfter(oldestMonitoring) ? requestedStart : oldestMonitoring;
+        final LocalDateTime end = Optional.ofNullable(to).orElseGet(DateUtils::nowUTC);
+        final Duration minDuration = minimalDuration();
+        final Duration adjustedDuration = interval.compareTo(minDuration) < 0
+                                          ? minDuration
+                                          : interval;
+        final List<MonitoringStats> monitoringStats = getStats(nodeName, start, end, adjustedDuration);
+        final MonitoringStatsWriter statsWriter = new MonitoringStatsWriter();
+        try {
+            return new StringInputStream(statsWriter.convertStatsToCsvString(monitoringStats));
+        } catch (IOException e) {
+            throw new IllegalStateException("Written csv stats file encoding differs from UTF-8.", e);
```

Comment:
Move error message to `MessageConstants`

Reply:
Done